app/lib/actions/katello/repository/update.rb

```diff
@@ -83,6 +91,27 @@ def handle_alternate_content_sources(repository, create_acs, delete_acs, old_url
           end
         end
 
+        def handle_acs_product_removal(repository)
```

Comment:
can we call this from inside handle_alternate_content_sources inside the `elsif delete_acs` block.

Also, why are we checking this and adding it to content type list, delete_acs being true implies this repo has no url right?
: 
```
if (repository.id == test_repo.id) && repository.url.present?
              repo_content_types.add(repository.content_type)
```

Reply:
I'll also look into if we can put the function call back where you're suggesting. There have been changes to the tests since I made that move so it's possible everything is kosher now.

Reply:
I did some testing and it looks like we're all good to put this logic back in the sequence. I believe making the removal of the product an action was what was messing things up. Just pushed fixes, going to mark as resolved.